### DISCUSSION TITLE
ITMS-90683: Missing purpose string in Info.plist

### DISCUSSION THREAD

**Discussion by kasparv** [airsdk/Adobe-Runtime-Support#2372] [General]
I have Mac OS Monterey 12.6.2 and XCode 14.2 (14C18), I tried using AIR SDK 50.0.1.3 and now 50.1.1.2 to build .ipa files and I'm still getting this message from Apple after uploading the .ipa files to App Store Connect:

> ITMS-90683: Missing purpose string in Info.plist - Your app’s code references one or more APIs that access sensitive user data, or the app has one or more entitlements that permit such access. The Info.plist file for the “Your.app” bundle should contain a NSLocationAlwaysAndWhenInUseUsageDescription key with a user-facing purpose string explaining clearly and completely why your app needs the data. If you’re using external libraries or SDKs, they may reference APIs that require a purpose string. While your app might not use these APIs, a purpose string is still required. For details, visit: https://developer.apple.com/documentation/uikit/protecting_the_user_s_privacy/requesting_access_to_protected_resources.

I'm reading that others may have similar issues - has anyone figured out a fix for this?

**Comment by ajwfrost** [comment]
You can add this as a temporary workaround using the `InfoAdditions` section of your application descriptor..
e.g.
```
    <iPhone> 
        <InfoAdditions><![CDATA[ 

			<key>NSLocationAlwaysAndWhenInUseUsageDescription</key>
			<string>We need to have your location information</string>

			<key>UIStatusBarStyle</key> 
			<string>UIStatusBarStyleBlackOpaque</string> 
			<key>UIDeviceFamily</key>
			<array>
				<!-- iPhone support -->
				<string>1</string> 
				<!-- iPad support -->
				<!--<string>2</string>-->
			</array>
        ]]></InfoAdditions> 
		<requestedDisplayResolution>high</requestedDisplayResolution>
    </iPhone> 
```

Currently we are automatically adding some of these usage purpose strings into the info.plist files (but not yet `NSLocationAlwaysAndWhenInUseUsageDescription`), the plan is to have defaults added here unless they're provided in the app descriptor section above.

thanks

**Reply by kasparv** [reply]
Thanks for the quick reply and workaround info!
My app does not actually use any location data or relevant APIs - is it something that the AIR SDK does?

**Reply by ajwfrost** [reply]
Yes I think they can just detect the fact that the AIR runtime has the geolocation APIs and so wants to ensure you have the appropriate strings available. To actually trigger them there's an API call that's needed later on..
